From 823c7c1ec79a88a1d7dab03b160f5fe79af17e71 Mon Sep 17 00:00:00 2001
From: Alex Hung <alex.hung@canonical.com>
Date: Tue, 17 Jun 2014 12:04:53 -0700
Subject: [PATCH] cpu: microcode: remove failures when kernel does not have
 newer version (LP: #1322534)

New systems usually have new microcode than kernel does, and
therefore reporting failures is not correct. This patch changes
it to report skipped when kernel does not have microcode updates.

Signed-off-by: Alex Hung <alex.hung@canonical.com>
Acked-by: Colin Ian King <colin.king@canonical.com>
Acked-by: Ivan Hu <ivan.hu@canonical.com>
---
 src/cpu/microcode/microcode.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/cpu/microcode/microcode.c b/src/cpu/microcode/microcode.c
index bf0dd928b6d9..d49316892cd8 100644
--- a/src/cpu/microcode/microcode.c
+++ b/src/cpu/microcode/microcode.c
@@ -230,13 +230,10 @@ static int microcode_test1(fwts_framework *fw)
 				}
 
 				/*
-				 * We found the old revision but not a
-				 * new revsion, failed
+				 * Kernel does not have newer version than BIOS
 				 */
 				if (info->new_revision == UNKNOWN) {
-					failed++;
-					fwts_failed(fw, LOG_LEVEL_MEDIUM, "MicrocodeNotUpdated",
-						"The kernel did not report that CPU %d has had a microcode update. "
+					fwts_log_info(fw, "The kernel did not report that CPU %d has had a microcode update. "
 						"The current firmware is revision 0x%x and probably has not been updated.",
 						cpu, info->old_revision);
 					continue;
-- 
1.9.3

